extractor/src/main/scala/com/digitalasset/extractor/config/ExtractorConfig.scala

```diff
@@ -40,8 +40,12 @@ object ExtractorConfig {
 }
 
 final case class TemplateConfig(moduleName: String, entityName: String)
-
-object TemplateConfig {
-  implicit val templateConfigOrdering: Ordering[TemplateConfig] =
-    Ordering.by(TemplateConfig.unapply)
+    extends Ordered[TemplateConfig] {
+
+  override def compare(that: TemplateConfig): Int = {
+    val a: (String, String) = (this.moduleName, this.entityName)
+    val b: (String, String) = (that.moduleName, that.entityName)
+    val O = implicitly[Ordering[(String, String)]]
+    O.compare(a, b)
```

Comment:
```suggestion
    Ordering.by(TemplateConfig.unapply).compare(this, that)
```

Reply:
nope... it will be slower... it will be creating a new instance of the Ordering for ever `compare` call:
```
  def by[T, S](f: T => S)(implicit ord: Ordering[S]): Ordering[T] = new Ordering[T] {
    def compare(x: T, y: T) = ord.compare(f(x), f(y))
    override def lt(x: T, y: T): Boolean = ord.lt(f(x), f(y))
    override def gt(x: T, y: T): Boolean = ord.gt(f(x), f(y))
    override def gteq(x: T, y: T): Boolean = ord.gteq(f(x), f(y))
    override def lteq(x: T, y: T): Boolean = ord.lteq(f(x), f(y))
  }
```

Reply:
Then stash it away somewhere. You've written the contramap application out by hand twice, which risks divergence later.

Reply:
you lost me. where is it going to diverge if the it is implemented:
```
  override def compare(that: TemplateConfig): Int = {
    val a: (String, String) = (this.moduleName, this.entityName)
    val b: (String, String) = (that.moduleName, that.entityName)
    val O = implicitly[Ordering[(String, String)]]
    O.compare(a, b)
  }
```

Reply:
Someone can edit the definition of `a` or `b` in a way inconsistent with the other; I don't have faith in my future self's ability to "be careful" about such mistakes.